### ISSUE TITLE
Fix up FO_AIS tests and get them to run with IOPX

### ISSUE THREAD

**Issue by ikalash** [sandialabs/Albany#761]
Request by @rstumin .  Was hoping to get help from @mperego and @bartgol to fix the input options so these tests run again.  Code complains about ```Field Origin: Output``` field.

**Comment by mperego** [comment]
I think I fixed the tests. 
However, I'm wondering if requiring Iopx library for using serial mesh is the right thing to do. On my workstation I don't have Iopx but "Use Serial Mesh: true" works fine. @bartgol @jewatkins 

@ikalash courios about the ProSPect$ label... we haven't used it in years...

**Comment by ikalash** [comment]
Thanks @mperego I'll close.  No idea where the ProSPect$ label came from.  I was curious about the $ symbol, but maybe it means "prospect money" (things funded by prospect)?

**Comment by ikalash** [comment]
@mperego : do you know why tests that don't use FRoSch but have it in the input file are failing?  

https://testing.sandia.gov/cdash/viewBuildError.php?buildid=10240241

**Comment by mperego** [comment]
@jewatkins do you understand what's the issue here"
https://sems-cdash-son.sandia.gov/cdash/test/1844066

**Comment by mperego** [comment]
OK, so I "should" have fixed all the issues now, except the Cuda one.

**Comment by bartgol** [comment]
> I'm wondering if requiring Iopx library for using serial mesh is the right thing to do. On my workstation I don't have Iopx but "Use Serial Mesh: true" works fine.

Isn't Iopx used to run decomp and/or to load a partitioned mesh? I wonder if it's not necessary for serial mesh which are partitioned at runtime...

**Comment by mperego** [comment]
> > I'm wondering if requiring Iopx library for using serial mesh is the right thing to do. On my workstation I don't have Iopx but "Use Serial Mesh: true" works fine.
> 
> Isn't Iopx used to run decomp and/or to load a partitioned mesh? I wonder if it's not necessary for serial mesh which are partitioned at runtime...

right, so what's the requirement for "Use Serial Mesh" to work? Based on the dashboard, it seems that the build on Camobap
https://sems-cdash-son.sandia.gov/cdash/viewTest.php?onlyfailed&buildid=25282
cannot handle "Use Serial Mesh". (In this test "Use Serial Mesh" was wrongly set to true independently of the check on Seacas and Iopx).

**Comment by jewatkins** [comment]
Doesn't seacas do the decomp at runtime?

**Comment by mperego** [comment]
> Doesn't seacas do the decomp at runtime? Maybe try turning that off to see if it fails?

Isn't Seacas required for the "decomp" function? Decomp works on Camobap, so if that's the case, checking for Seacas is not enough for guarantiing that "Use Serial Mesh" works

**Comment by jewatkins** [comment]
okay, what about Ioss? I know that's needed.

**Comment by mperego** [comment]
Maybe we need to check if NetCDF library is built with PNetCDF support:

This is the error on Camobap
```
 Warning/Error: [ex_open_par_int]
	EXODUS: ERROR: Attempting to open the NetCDF file:
	'antarctica_2d.exo'
	The NetCDF library was not built with PNetCDF support as required for parallel access to this file.

	NetCDF: Unknown file format
```

**Comment by jewatkins** [comment]
maybe in cmake, check if something like this gets pulled: `TPL_Netcdf_Enables_PNetcdf`

**Comment by mperego** [comment]
> @jewatkins do you understand what's the issue here: https://sems-cdash-son.sandia.gov/cdash/test/1844066

B.t.w. @jewatkins have you seen this. This is different from the exodus issue.

**Comment by ikalash** [comment]
I don't have PNetCDF on camobap.  I build my own libs there and there are some problems with building netcdf with pnetcdf.  I would say the code should not barf even if the user does not have pnetcdf - I'm not sure why it does.

**Comment by jewatkins** [comment]
> > @jewatkins do you understand what's the issue here: https://sems-cdash-son.sandia.gov/cdash/test/1844066
> 
> B.t.w. @jewatkins have you seen this. This is different from the exodus issue.

I have not seen that. There's probably a compatibility issue now when using Kokkos::Cuda + MueLu without kokkos refactor on. The performance tests use kokkos so that's why they're working still. It's worth discussing with other muelu folks to see if that's the desired behavior.

**Comment by jewatkins** [comment]
If it is the desired behavior, will have to create a second test with the kokkos syntax and turn off this test with cuda builds.

**Comment by ikalash** [comment]
I don't understand why "Use Serial Mesh" would not work on camobap whereas it would on other platforms...  maybe a special configuration of netcdf is needed?  I can try to look into this but it will not be immediately.  I'm curious also if this problem happens on @lxmota 's machines, which are identical to camobap in terms of OS.  I can try building Albany there to check.

**Comment by mperego** [comment]
@ikalash It does not work there because you don't have parallel netcdf or hdf5 such that exodus can import a file in parallel. I'm working on a fix.

**Comment by mperego** [comment]
> > > @jewatkins do you understand what's the issue here: https://sems-cdash-son.sandia.gov/cdash/test/1844066
> > 
> > 
> > B.t.w. @jewatkins have you seen this. This is different from the exodus issue.
> 
> I have not seen that. There's probably a compatibility issue now when using Kokkos::Cuda + MueLu without kokkos refactor on. The performance tests use kokkos so that's why they're working still. It's worth discussing with other muelu folks to see if that's the desired behavior.

@jewatkins Would you be willing of bringing this up with the MueLu folks?

**Comment by mperego** [comment]
@jewatkins Now that we fixed the issue with exodus, the MueLu tests seem to fail  also on OpenMP builds. 

```
p=0: *** Caught standard std::exception of type 'Teuchos::bad_any_cast' :

 /nightlyAlbanyTests/Results/Trilinos/packages/muelu/src/MueCentral/MueLu_VariableContainer.hpp:101:
 
 Throw number = 1
 
 Throw test that evaluated to true: data_->type() != typeid(T)
 
 Error, cast to type Data<Teuchos::RCP<MueLu::GraphBase<int, long long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > >> failed since the actual underlying type is 'Teuchos::RCP<MueLu::LWGraph_kokkos<int, long long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP, Kokkos::HostSpace> > >!
```

**Comment by jewatkins** [comment]
That looks like the same issue. Yes, I can post the issue.

**Comment by jewatkins** [comment]
Done. https://github.com/trilinos/Trilinos/issues/9943

Now that I have thought about this some more, I think I do remember that error message. Even if muelu were to work for openmp/cuda, it would probably try to do everything in serial. So we would need to create a new test for openmp/cuda builds anyways (or just modify the test so that it uses kokkos for everything).

**Comment by jewatkins** [comment]
I'll check tomorrow to see if turning off kokkos refactor fixes the issue for openmp/cuda.

**Comment by mperego** [comment]
@jewatkins thanks for fixing this!
